src/Geom_Poly.f90

```diff
@@ -411,7 +411,9 @@ SUBROUTINE calcCentroid(this)
     ENDDO
   ENDIF
   CALL this%centroid%clear()
-  CALL this%centroid%init(DIM=2,X=xcent/this%area,Y=ycent/this%area)
+  IF(this%area > 0.0_SRK) THEN
+    CALL this%centroid%init(DIM=2,X=xcent/this%area,Y=ycent/this%area)
+  ENDIF
```

Comment:
But what if it's not > 0?

Reply:
Yeah I guess it could either be the be the centroid of a point, in which case  maybe this is appropriate
```
IF(this%area > 0.0_SRK) THEN
  CALL this%centroid%init(DIM=2,X=xcent/this%area,Y=ycent/this%area)
ELSE
  CALL this%cnetroid%init(DIM=2,X=xcent,Y=ycent)
ENDIF
```
or the centroid of negative area, which should probably throw an error.

Reply:
that seems like a reasonable approach